src/game/common/system/ramfile.cpp

```diff
@@ -303,28 +267,31 @@ bool RAMFile::Open_From_Archive(File *file, Utf8String const &name, int pos, int
         return false;
     }
 
-    if (File::Open(name.Str(), READ | BINARY)) {
-        if (m_data != nullptr) {
-            delete[] m_data;
-            m_data = nullptr;
-        }
+    if (!File::Open(name.Str(), READ | BINARY)) {
+        return false;
+    }
 
-        m_size = size;
-        m_data = new char[size];
+    if (m_data != nullptr) {
+        delete[] m_data;
+        m_data = nullptr;
+    }
 
-        if (file->Seek(pos, START) == pos) {
-            if (file->Read(m_data, m_size) == size) {
-                m_filename = file->Get_File_Name();
+    m_data = new char[size];
+    m_size = size;
 
-                return true;
-            }
-        }
+    if (!file->Seek(pos, START) == pos) {
+        return false;
+    }
+
+    if (!file->Read(m_data, m_size) == size) {
+        return false;
```

Comment:
`m_data` does not clean up here. Is this intentional? See `RAMFile::Open` function above, cleans up after itself.

Reply:
This is correct to the original.